webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +72,86 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
-        throw new Error(err)
+        this.$toast.error(err.message)
       }
     },
+    hoverUpdate() {
+      if (this.hover) return
+
+      // update is triggered kind of one second before the user clicks
+      this.hover = true
+      this.updateNotifications()
+    },
+    callToggleMenu(toggleMenu) {
+      if (this.menuTimerId) return
+
+      this.updateNotifications() // then the update works even on mobile by click and without hover
+      // open menu after update is done, but latest after delay of "NOTIFICATIONS_MENU_OPEN_LATEST_INTERVAL"
+      this.menuDelayTicks = 0
+      this.menuTimerId = setInterval(() => {
+        if (!this.updateOn || NOTIFICATIONS_MENU_OPEN_LATEST_INTERVAL <= this.menuDelayTicks) {
+          toggleMenu()
+          clearInterval(this.menuTimerId)
+          this.menuTimerId = null
+        } else this.menuDelayTicks += NOTIFICATIONS_CHECK_UPDATED_INTERVAL
+      }, NOTIFICATIONS_CHECK_UPDATED_INTERVAL)
+    },
+    equalNotification(a, b) {
+      return a.from.id === b.from.id && a.createdAt === b.createdAt && a.reason === b.reason
+    },
   },
   computed: {
-    totalNotifications() {
-      return (this.notifications || []).length
+    totalNotificationsCount() {
+      return (this.displayedNotifications || []).length
+    },
+    unreadNotificationsCount() {
+      let countUnread = 0
+      if (this.displayedNotifications) {
+        this.displayedNotifications.forEach(notification => {
+          if (!notification.read) countUnread++
+        })
+      }
+      return countUnread
     },
   },
   apollo: {
     notifications: {
       query() {
         return notificationQuery(this.$i18n)
       },
+      pollInterval() {
+        return NOTIFICATIONS_POLL_INTERVAL
+      },
+      update(data) {
+        const newNotifications = data.notifications.filter(newN => {
+          return !this.displayedNotifications.find(oldN => this.equalNotification(newN, oldN))
+        })
+        this.displayedNotifications = newNotifications
+          .concat(this.displayedNotifications)
+          .sort((a, b) => {
+            return a.createdAt === b.createdAt
```

Comment:
Why not:
```suggestion
  return new Date(a.createdAt) - new Date(b.createdAt) // maybe reverse a with b, I'm not sure
```

Reply:
Thanks for the hint @roschaefer ! 😍
Done